sema/errors.go

```diff
@@ -4739,6 +5439,32 @@ func (e *DefaultDestroyInvalidParameterError) Error() string {
 	return fmt.Sprintf("`%s` is not a valid parameter type for a default destroy event", e.ParamType.QualifiedString())
 }
 
+func (e *DefaultDestroyInvalidParameterError) SecondaryError() string {
+	return "default destroy events only support storable types as parameters"
```

Comment:
Sorry to jump in here again, but I just realized that the requirement is different, after checking the code (https://github.com/onflow/cadence/blob/c149e8f1654955e5a75db7de281845dea9cd6040/sema/check_composite_declaration.go#L2193-L2200): The parameter types must be primitive types (this might actually also not be documented well, not sure)

```suggestion
	return "default destroy events only support primitive types as parameters"
```